addons/website_sale_coupon/models/sale_order.py

```diff
@@ -93,3 +113,29 @@ def _gc_abandoned_coupons(self, *args, **kwargs):
             coupon.sales_order_id.applied_coupon_ids -= coupon
         coupon_to_reset.write({'state': 'new'})
         coupon_to_reset.mapped('sales_order_id').recompute_coupon_lines()
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    def unlink(self):
+        reward_lines = self.filtered(lambda line: line.is_reward_line)
+        status = request.session.get('pending_coupon') or {}
+        code = status.get('code')
+
+        if request and reward_lines and code:
+            pending_coupon = self.env['coupon.coupon'].sudo().search([
+                ('code', '!=', False),
```

Comment:
This is useless, this case is already covered by the other condition `('code', '=', code)`..

Reply:
`code` is a required field anyway.. it can't be false